bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/handler/GenericEventTriggerHandler.java

```diff
@@ -52,13 +54,15 @@
     private static final String CFG_EVENT_TOPIC = "eventTopic";
     private static final String CFG_EVENT_SOURCE = "eventSource";
     private static final String CFG_EVENT_TYPES = "eventTypes";
+    private static final String CFG_CHANNEL_EVENT = "eventOnChannel";
```

Comment:
This properties is specific for the channel events and there is no place for it in this class. If you want more specific handler it is better to extends this one and add your property and apply it to the Then the core.ChannelEventTrigger.  
Actually GenericEventTriggerHandler implements EventFilter which at the moment just check this:
    @Override
    public boolean apply(Event event) {
        logger.trace("->FILTER: {}:{}", event.getTopic(), source);
        return event.getTopic().contains(source);
    }
May be better approach is to add filter configuration property and change the apply method to check for it.
For example:
  public boolean apply(Event event) {
        logger.trace("->FILTER: {}:{}", event.getTopic(), source);
         String topic = event.getTopic();
        return topic.contains(source) && (filter== null || topic.contains(filter));
    }
where the filter can be your eventOnChannel
Is this approach suitable is for you?